--- smbldap-tools-0.8.8-orig/Makefile	2005-04-03 18:03:34.000000000 +0200
+++ smbldap-tools-0.8.8/Makefile	2005-04-18 18:02:48.000000000 +0200
@@ -24,11 +24,11 @@
 all:	prep distclean rpm
 
 install:
-	@mkdir -p $(sbindir)
-	@mkdir -p $(sysconfdir)/smbldap-tools/
-	$(install) -m0755 smbldap-* smbldap_tools.pm $(sbindir)
-	$(install) -m0644 smbldap.conf $(sysconfdir)/smbldap-tools/
-	$(install) -m0600 smbldap_bind.conf $(sysconfdir)/smbldap-tools/
+	@mkdir -p $(DESTDIR)/$(sbindir)
+	@mkdir -p $(DESTDIR)/$(sysconfdir)/smbldap-tools/
+	$(install) -m0755 smbldap-* smbldap_tools.pm $(DESTDIR)/$(sbindir)
+	$(install) -m0644 smbldap.conf $(DESTDIR)/$(sysconfdir)/smbldap-tools/
+	$(install) -m0600 smbldap_bind.conf $(DESTDIR)/$(sysconfdir)/smbldap-tools/
 
 clean:
 	$(rm) -f smb.conf smbldap.conf
