diff -ur smbldap-tools-0.9.6/smbldap.conf smbldap-tools-0.9.6-krb5/smbldap.conf
--- smbldap-tools-0.9.6/smbldap.conf	2008-08-04 23:38:43.000000000 +0200
+++ smbldap-tools-0.9.6-krb5/smbldap.conf	2009-06-22 16:51:28.927035176 +0200
@@ -214,6 +214,14 @@
 
 ##############################################################################
 #
+# Kerberos Configuration
+#
+##############################################################################
+
+# KERBEROS_REALM="EXAMPLE.COM"
+
+##############################################################################
+#
 # SMBLDAP-TOOLS Configuration (default are ok for a RedHat)
 #
 ##############################################################################
Only in smbldap-tools-0.9.6-krb5: smbldap.conf~
diff -ur smbldap-tools-0.9.6/smbldap-useradd smbldap-tools-0.9.6-krb5/smbldap-useradd
--- smbldap-tools-0.9.6/smbldap-useradd	2009-06-22 16:19:38.853899020 +0200
+++ smbldap-tools-0.9.6-krb5/smbldap-useradd	2009-06-22 16:37:34.637118213 +0200
@@ -547,21 +547,43 @@
             $valacctflags = "$tmp";
         }
 
-        my $modify = $ldap_master->modify(
-            "uid=$userName,$config{usersdn}",
-            changes => [
-                add => [ objectClass        => 'sambaSAMAccount' ],
-                add => [ sambaPwdLastSet    => "$valpwdlastset" ],
-                add => [ sambaLogonTime     => '0' ],
-                add => [ sambaLogoffTime    => '2147483647' ],
-                add => [ sambaKickoffTime   => '2147483647' ],
-                add => [ sambaPwdCanChange  => "$valpwdcanchange" ],
-                add => [ sambaPwdMustChange => "$valpwdmustchange" ],
-                add => [ displayName        => "$displayName" ],
-                add => [ sambaAcctFlags     => "$valacctflags" ],
-                add => [ sambaSID           => "$config{SID}-$userRid" ]
-            ]
-        );
+	if ( defined( $KERBEROS_REALM ) ) {
+		my $modify = $ldap_master->modify(
+		    "uid=$userName,$config{usersdn}",
+		    changes => [
+			add => [ objectClass        => 'sambaSAMAccount' ],
+			add => [ sambaPwdLastSet    => "$valpwdlastset" ],
+			add => [ sambaLogonTime     => '0' ],
+			add => [ sambaLogoffTime    => '2147483647' ],
+			add => [ sambaKickoffTime   => '2147483647' ],
+			add => [ sambaPwdCanChange  => "$valpwdcanchange" ],
+			add => [ sambaPwdMustChange => "$valpwdmustchange" ],
+			add => [ displayName        => "$displayName" ],
+			add => [ sambaAcctFlags     => "$valacctflags" ],
+			add => [ sambaSID           => "$config{SID}-$userRid" ],
+			add => [ objectClass        => 'krb5Principal' ],
+			add => [ objectClass        => 'krb5KDCEntry' ],
+			add => [ krb5PrincipalName  => "$userName\@$KERBEROS_REALM" ],
+			add => [ krb5KeyVersionNumber => '0' ]
+		    ]
+		);
+	} else {
+		my $modify = $ldap_master->modify(
+		    "uid=$userName,$config{usersdn}",
+		    changes => [
+			add => [ objectClass        => 'sambaSAMAccount' ],
+			add => [ sambaPwdLastSet    => "$valpwdlastset" ],
+			add => [ sambaLogonTime     => '0' ],
+			add => [ sambaLogoffTime    => '2147483647' ],
+			add => [ sambaKickoffTime   => '2147483647' ],
+			add => [ sambaPwdCanChange  => "$valpwdcanchange" ],
+			add => [ sambaPwdMustChange => "$valpwdmustchange" ],
+			add => [ displayName        => "$displayName" ],
+			add => [ sambaAcctFlags     => "$valacctflags" ],
+			add => [ sambaSID           => "$config{SID}-$userRid" ]
+		    ]
+		);
+	}
 
         $modify->code && die "failed to add entry: ", $modify->error;
 
