diff -ur smbldap-tools-0.9.6/smbldap.conf smbldap-tools-0.9.6-krb5/smbldap.conf
--- smbldap-tools-0.9.6/smbldap.conf	2008-08-04 23:38:43.000000000 +0200
+++ smbldap-tools-0.9.6-krb5/smbldap.conf	2009-06-22 16:51:28.927035176 +0200
@@ -214,6 +214,14 @@
 
 ##############################################################################
 #
+# Kerberos Configuration
+#
+##############################################################################
+
+# KERBEROS_REALM="EXAMPLE.COM"
+
+##############################################################################
+#
 # SMBLDAP-TOOLS Configuration (default are ok for a RedHat)
 #
 ##############################################################################
diff -ur smbldap-tools-0.9.6/smbldap-useradd.pl smbldap-tools-0.9.6-krb5/smbldap-useradd.pl
--- smbldap-tools-0.9.6/smbldap-useradd.pl	2009-06-22 16:19:38.853899020 +0200
+++ smbldap-tools-0.9.6-krb5/smbldap-useradd.pl	2009-06-22 16:37:34.637118213 +0200
@@ -405,6 +405,13 @@
     exit 0;
 }
 
+my $userPass;
+if ( $config{hash_encrypt} eq "K5KEY" ) {
+	$userPass = "{K5KEY}";
+} else {
+	$userPass = "{crypt}x";
+}
+
 # USER ACCOUNT
 # add posix account first
 my @objectclass = qw(top person organizationalPerson posixAccount);
@@ -418,7 +425,7 @@
     'homeDirectory' => $userHomeDirectory,
     'loginShell'    => $config{userLoginShell},
     'gecos'         => $config{userGecos},
-    'userPassword'  => "{crypt}x"
+    'userPassword'  => "$userPass"
 );
 
 push(@objectclass, 'shadowAccount') if ($config{shadowAccount});
@@ -547,21 +547,47 @@
             $valacctflags = "$tmp";
         }
 
-        my $modify = $ldap_master->modify(
-            "uid=$userName,$config{usersdn}",
-            changes => [
-                add => [ objectClass        => 'sambaSAMAccount' ],
-                add => [ sambaPwdLastSet    => "$valpwdlastset" ],
-                add => [ sambaLogonTime     => '0' ],
-                add => [ sambaLogoffTime    => '2147483647' ],
-                add => [ sambaKickoffTime   => '2147483647' ],
-                add => [ sambaPwdCanChange  => "$valpwdcanchange" ],
-                add => [ sambaPwdMustChange => "$valpwdmustchange" ],
-                add => [ displayName        => "$displayName" ],
-                add => [ sambaAcctFlags     => "$valacctflags" ],
-                add => [ sambaSID           => "$config{SID}-$userRid" ]
-            ]
-        );
+	my $modify;
+	if ( defined( $config{KERBEROS_REALM} ) ) {
+		$modify = $ldap_master->modify(
+		    "uid=$userName,$config{usersdn}",
+		    changes => [
+			add => [ objectClass        => 'sambaSAMAccount' ],
+			add => [ sambaPwdLastSet    => "$valpwdlastset" ],
+			add => [ sambaLogonTime     => '0' ],
+			add => [ sambaLogoffTime    => '2147483647' ],
+			add => [ sambaKickoffTime   => '2147483647' ],
+			add => [ sambaPwdCanChange  => "$valpwdcanchange" ],
+			add => [ sambaPwdMustChange => "$valpwdmustchange" ],
+			add => [ displayName        => "$displayName" ],
+			add => [ sambaAcctFlags     => "$valacctflags" ],
+			add => [ sambaSID           => "$config{SID}-$userRid" ],
+			add => [ objectClass        => 'krb5Principal' ],
+			add => [ objectClass        => 'krb5KDCEntry' ],
+			add => [ krb5PrincipalName  => "$userName\@$config{KERBEROS_REALM}" ],
+			add => [ krb5KeyVersionNumber => '0' ],
+			add => [ krb5KDCFlags       => '126'],
+			add => [ krb5MaxRenew       => '604800'],
+			add => [ krb5MaxLife        => '86400']
+		    ]
+		);
+	} else {
+		$modify = $ldap_master->modify(
+		    "uid=$userName,$config{usersdn}",
+		    changes => [
+			add => [ objectClass        => 'sambaSAMAccount' ],
+			add => [ sambaPwdLastSet    => "$valpwdlastset" ],
+			add => [ sambaLogonTime     => '0' ],
+			add => [ sambaLogoffTime    => '2147483647' ],
+			add => [ sambaKickoffTime   => '2147483647' ],
+			add => [ sambaPwdCanChange  => "$valpwdcanchange" ],
+			add => [ sambaPwdMustChange => "$valpwdmustchange" ],
+			add => [ displayName        => "$displayName" ],
+			add => [ sambaAcctFlags     => "$valacctflags" ],
+			add => [ sambaSID           => "$config{SID}-$userRid" ]
+		    ]
+		);
+	}
 
         $modify->code && die "failed to add entry: ", $modify->error;
 
--- smbldap-tools-0.9.6/smbldap-passwd.pl.orig	2009-06-25 22:40:05.764901892 +0200
+++ smbldap-tools-0.9.6/smbldap-passwd.pl	2009-06-25 22:49:03.906899121 +0200
@@ -77,6 +77,10 @@
     }
 }
 
+if ($config{hash_encrypt} eq "K5KEY" ) {
+    die "Refusing to mess with Kerberos passwords/keys,\nuse smbpasswd, kpasswd or pam passwd\n";
+}
+
 if (!defined($user)) {
     $user = getpwuid($<);		# $user=$ENV{"USER"};
 }
